application/use_cases/stock_items/delete_stock_item/delete_stock_item_entity_existence_checker.py

```diff
@@ -0,0 +1,17 @@
+from clapy import EntityExistenceChecker
+
+from application.services.ientity_existence_checker import IEntityExistenceChecker
+from application.use_cases.stock_items.delete_stock_item.delete_stock_item_input_port import DeleteStockItemInputPort
+from application.use_cases.stock_items.delete_stock_item.idelete_stock_item_output_port import IDeleteStockItemOutputPort
+from domain.entities.stock_item import StockItem
+
+
+class DeleteStockItemEntityExistenceChecker(EntityExistenceChecker):
+
+    def __init__(self, existence_checker: IEntityExistenceChecker):
+        self.existence_checker = existence_checker
+
+    async def execute_async(self, input_port: DeleteStockItemInputPort, output_port: IDeleteStockItemOutputPort):
+        if not self.existence_checker.does_entity_exist(StockItem, input_port.stock_item_id):
+            self.has_failures = True
+            await output_port.present_stock_item_not_found_async(input_port.stock_item_id)
```

Comment:
so i got this idea from James (and i may have implemented it wrong), but the idea being you have an entity ID that encapsulates the idea of an identity for domain entities, and hides the actual implementation of the identity from the domain and application. It allows us to easily change the implementation of the identity without affecting the inner layers. The only layer that cares what it is and must translate it is the framework/persistence, e.g. the API must spit out the UUID instead of the EntityID as EntityID means nothing outside this domain, and it must also translate an incoming UUID for the application layer. Another example is the persistence must be able to translate the identity of entities into the identity for the models that are persisted.

Reply:
Very interesting.
Abstracting the identity/identifier is not something i considered.

Do we do this at work?
Never